src/puppetlabs/puppetdb/http/command.clj

```diff
@@ -1,19 +1,45 @@
 (ns puppetlabs.puppetdb.http.command
   (:require [puppetlabs.puppetdb.command.constants :refer [command-names]]
             [puppetlabs.trapperkeeper.core :refer [defservice]]
+            [clojure.string :as str]
             [clojure.tools.logging :as log]
             [puppetlabs.puppetdb.command :as command]
             [puppetlabs.puppetdb.http :as http]
+            [puppetlabs.puppetdb.cheshire :as json]
             [net.cgrand.moustache :as moustache]
             [puppetlabs.puppetdb.middleware :as mid]
             [compojure.core :as compojure]))
 
+(def min-supported-commands
+  {"replace catalog" 6
+   "replace facts" 4
+   "store report" 5
+   "deactivate node" 3})
+
+(def valid-commands-str (str/join ", " (sort (vals command-names))))
+
 (defn enqueue-command-handler
   "Enqueues the command in request and returns a UUID"
   [connection endpoint]
   (fn [{:keys [body-string] :as request}]
-    (let [uuid (command/enqueue-raw-command! connection endpoint body-string)]
-      (http/json-response {:uuid uuid}))))
+    (let [{:keys [command version]} (json/parse-string body-string true)
+          min-supported (get min-supported-commands command ::invalid)]
+      (cond
+        (= ::invalid min-supported)
+        (http/json-response
```

Comment:
that doesn't return JSON. We use it in places but I think we should swap it out. There's a ticket with links to the company guidelines here: https://tickets.puppetlabs.com/browse/PDB-1511